### PR TITLE
Files/Nautilus: Convert to pdf from context menu

### PR DESCRIPTION
**Pull request by langfingaz**
Until now the tips section is empty and I thought that this script would be a nice addition for GNOME/Nautilus users:

With `xopp-to-pdf` in the Nautilus scripts folder one can quickly select multiple Xournal++ files and convert them to pdf.

What do you think about this addition. Is it to specific (e.g. just for a too small user group)? Or does it sound like a good addition?

I was not sure about where to place the image and script files. I chose `img` for the screenshot and created a folder `resources` with the script inside. If you prefer another folder structure, this could for sure be rearranged.

### CODE REVIEW

**Comment by LittleHuba** [comment]
Nice idea. I think it is okay to add this. 

I would modify CMake to install it into the correct directory right away for Linux targets. This way it is available for users with Nautilus straightaway and should not bother anybody that does not have it.

Also the package manager would then take care of it in contrast to manually copying the file.

**Comment by langfingaz** [comment]
Great to hear that positive feedback. Adding it to CMake would make sense for sure, however I can't help as much with that since I have no experience with CMake.

But I have just extended the script a bit to write messages to stderr if notify-send is not available/installed.

**Comment by LittleHuba** [comment]
Just to make sure that there is no hidden bug:
If I have a file `/home/user/a.xopp` and convert it with your script, is the new name `/home/user/a.xopp.pdf`?

**Comment by LittleHuba** [comment]
For direct integration with Xournal++ we would need to move this PR over to the repository of the application. The CMake routine is quite easy. What you want is [this](https://cmake.org/cmake/help/latest/command/install.html#files).

**Comment by Technius** [comment]
How would this integrate with system-wide installations? I can't install this for a specific user if I want to generate a .deb or tarball.

**Comment by LittleHuba** [comment]
Instead of `/home/user/.local/share/...` The path is then normally `/user/share/...`.

Unfortunately I do not have Nautilus installed and can not check this myself. @langfingaz could you make sure this path exists and check if the script is handled correctly within Nautilus?

**Comment by rolandlo** [comment]
> Instead of `/home/user/.local/share/...` The path is then normally `/user/share/...`.
> 
> Unfortunately I do not have Nautilus installed and can not check this myself. @langfingaz could you make sure this path exists and check if the script is handled correctly within Nautilus?

You mean `/usr/share/...` . 
I have my scripts in `/home/<username>/.local/share/nautilus/scripts`, but there is no folder `/usr/share/nautilus` to begin with.
The website https://help.ubuntu.com/community/NautilusScriptsHowto also doesn't mentioned any other path than the first one.

**Comment by LittleHuba** [comment]
> You mean `/usr/share/...` .

Of course. Wrote that with my smartphone which seems to not like paths for Linux...

> there is no folder /usr/share/nautilus to begin with.

That is a bummer. Then automatic installation does not make sense. While theoretically possible it would be one big hack. So let's just include it on the webpage and be done with it.

File: docs/resources/xopp-to-pdf [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env bash
+
+SCRIPT_TITLE=xopp-to-pdf
```
Comment (LittleHuba) [inline docs/resources/xopp-to-pdf:3]: I would modify the title to be somewhat nicer to read.
```suggestion
SCRIPT_TITLE=Convert *.xopp to *.pdf
```

File: docs/resources/xopp-to-pdf [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  if [ "${file}" = '' ]; then
+    continue;
+  fi
+
+  out="${file}".pdf;
+
+  if [ -f "${out}" ]; then
+    notify "${SCRIPT_TITLE}" "The file ${out} does already exist! Skipping conversion of $(basename "${file}")" --icon=dialog-information;
+  else
+    xournalpp --create-pdf="${out}" "${file}" || notify "${SCRIPT_TITLE}" "Conversion of ${file} failed!" --icon=dialog-information;
```
Comment (LittleHuba) [inline docs/resources/xopp-to-pdf:44]: You should check if this file is actually a *.xopp before trying to convert it.

**Review by LittleHuba — CHANGES_REQUESTED** [review]

**Comment by Technius** [comment]
Hi @langfingaz, are you still interested in including this? If you are, please note that we recently updated the license terms.

File: docs/resources/xopp-to-pdf (hunk 1 above)
Comment (langfingaz) [reply docs/resources/xopp-to-pdf:3]: Good idea, I'll add it with single quotes.

File: docs/resources/xopp-to-pdf (hunk 2 above)
Comment (langfingaz) [reply docs/resources/xopp-to-pdf:44]: Good point, I'll add that as well.

**Comment by langfingaz** [comment]
Sorry for my late replies and thanks for the pingback @Technius. I'm still interested in this.

I as well couldn't find a good way to include the script in a global directory. One could place it into `/etc/skel` so that it is copied for new users created after that. But I doubt that this is useful for us.

**Comment by langfingaz** [comment]
> Just to make sure that there is no hidden bug:
> If I have a file `/home/user/a.xopp` and convert it with your script, is the new name `/home/user/a.xopp.pdf`?

Yes, this is intended. I thought of this workflow:
- A user annotates `slides.pdf` which is then (by default) saved to `slides.xopp`. The original PDF stays unchanged.
- Later it want's to export the annotated document back to PDF (e.g. to view it on a different device or share it). `slides.pdf` does already exist, but `slides.pdf.xopp` is an alternative filename that visualizes that the exported document is based on the `.xopp` file including the annotations.

Other proposals are welcome.

**Comment by langfingaz** [comment]
So, I resolved the proposed changes by @LittleHuba and hope that you like the added Nemo support. Let me know if there is anything else that needs further changes.

**Review by Technius — APPROVED** [review]
LGTM, thanks. I would suggest adding a copyright/description header to the top of the script to make sure others give you credit if they use/modify it.

**Comment by langfingaz** [comment]
> LGTM, thanks. I would suggest adding a copyright/description header to the top of the script to make sure others give you credit if they use/modify it.

I've added such a header. This project uses the MIT License for non text-documents which is fine for me and I've thus set the same license for my script.

**Comment by Technius** [comment]
Thanks.

**Pull request merged**